cmd/mtaBuild_generated.go

```diff
@@ -0,0 +1,197 @@
+package cmd
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/SAP/jenkins-library/pkg/config"
+	"github.com/SAP/jenkins-library/pkg/log"
+	"github.com/SAP/jenkins-library/pkg/piperenv"
+	"github.com/SAP/jenkins-library/pkg/telemetry"
+	"github.com/spf13/cobra"
+)
+
+type mtaBuildOptions struct {
+	BuildTarget         string `json:"buildTarget,omitempty"`
+	MtaBuildTool        string `json:"mtaBuildTool,omitempty"`
+	MtarName            string `json:"mtarName,omitempty"`
+	MtaJarLocation      string `json:"mtaJarLocation,omitempty"`
+	Extensions          string `json:"extensions,omitempty"`
+	Platform            string `json:"platform,omitempty"`
+	ApplicationName     string `json:"applicationName,omitempty"`
+	DefaultNpmRegistry  string `json:"defaultNpmRegistry,omitempty"`
+	ProjectSettingsFile string `json:"projectSettingsFile,omitempty"`
+	GlobalSettingsFile  string `json:"globalSettingsFile,omitempty"`
+}
+
+type mtaBuildCommonPipelineEnvironment struct {
+	mtarFilePath string
+}
+
+func (p *mtaBuildCommonPipelineEnvironment) persist(path, resourceName string) {
+	content := []struct {
+		category string
+		name     string
+		value    string
+	}{
+		{category: "", name: "mtarFilePath", value: p.mtarFilePath},
+	}
+
+	errCount := 0
+	for _, param := range content {
+		err := piperenv.SetResourceParameter(path, resourceName, filepath.Join(param.category, param.name), param.value)
+		if err != nil {
+			log.Entry().WithError(err).Error("Error persisting piper environment.")
+			errCount++
+		}
+	}
+	if errCount > 0 {
+		os.Exit(1)
+	}
+}
+
+// MtaBuildCommand Performs an mta build
+func MtaBuildCommand() *cobra.Command {
+	metadata := mtaBuildMetadata()
+	var stepConfig mtaBuildOptions
+	var startTime time.Time
+	var commonPipelineEnvironment mtaBuildCommonPipelineEnvironment
```

Comment:
I think we should avoid that name to not create confusion with the jenkins lib. Also "common" makes no sense here as the state is not shared amongst steps, right?

Reply:
>I think we should avoid that name to not create confusion with the jenkins lib

... and I thought it would be a good idea to have that naming in order to see similarities to the jenkins environment. In case we use the go lib from the Jenkins we [read](https://github.com/SAP/jenkins-library/blob/master/vars/commonPipelineEnvironment.groovy#L193) the data we persist here into the commonPipelineEnvironment and vice versa.

>as the state is not shared amongst steps, right?

Wrong. It is about sharing between steps. The mtar name is set here and used later on by the neoDeploy step. The whole struc was invented to support sharing between steps.